### ISSUE TITLE
[bug] conan info --paths with short_paths=True corrupts cache

### ISSUE THREAD

**Issue by sorekz** [conan-io/conan#8172]
### Environment Details (include every applicable attribute)
  * Operating System+version: Windows 10
  * Compiler+version: -
  * Conan version: 1.32.0
  * Python version: 3.7.1

### Steps to reproduce (Include if Applicable)
The problem occurs since conan 1.31.0 and probably has do to with #7900 but not sure. It seems to only occur with recipes that have short_path=True (not with use_alsways_short_paths=True) and when doing a `conan info ... --paths`. Without `--paths` its working normal. The first time the command is executed it will create the package and other folders and execute. When doing a conan info on the package again (or any other command that creates the dependency graph), it throws an error because the package folder already exists (it's not looking for the existance of the short paths folder). Anyway here's an example:

I couldn't find a recipe with short_paths=True but it can be reproduced by editing the poco.
- `conan info poco/1.9.4@`
- Open the export folder and edit the conanfile
   Add the attribute `short_paths= True` and save it
- `conan info poco/1.9.4@ --paths`
   After the command the package folder for the default profile exists
- `conan info poco/1.9.4@ --paths`
   Throws the error below

### Logs (Executed commands with output) (Include/Attach if Applicable)
```
conan info poco/1.9.4@ --paths
Traceback (most recent call last):
  File "C:\Users\Daniel\AppData\Roaming\Python\Python37\site-packages\conans\client\command.py", line 2115, in run
    method(args[0][1:])
  File "C:\Users\Daniel\AppData\Roaming\Python\Python37\site-packages\conans\client\command.py", line 747, in info
    lockfile=args.lockfile)
  File "C:\Users\Daniel\AppData\Roaming\Python\Python37\site-packages\conans\client\conan_api.py", line 94, in wrapper
    return f(api, *args, **kwargs)
  File "C:\Users\Daniel\AppData\Roaming\Python\Python37\site-packages\conans\client\conan_api.py", line 743, in info
    update, False, remotes, recorder)
  File "C:\Users\Daniel\AppData\Roaming\Python\Python37\site-packages\conans\client\graph\graph_manager.py", line 116, in load_graph
    apply_build_requires=apply_build_requires)
  File "C:\Users\Daniel\AppData\Roaming\Python\Python37\site-packages\conans\client\graph\graph_manager.py", line 244, in _resolve_graph
    graph_lock=graph_lock)
  File "C:\Users\Daniel\AppData\Roaming\Python\Python37\site-packages\conans\client\graph\graph_manager.py", line 363, in _load_graph
    apply_build_requires=apply_build_requires)
  File "C:\Users\Daniel\AppData\Roaming\Python\Python37\site-packages\conans\client\graph\graph_manager.py", line 284, in _recurse_build_requires
    self._binary_analyzer.evaluate_graph(graph, build_mode, update, remotes, nodes_subset, root)
  File "C:\Users\Daniel\AppData\Roaming\Python\Python37\site-packages\conans\client\graph\graph_binaries.py", line 384, in evaluate_graph
    self._evaluate_node(node, build_mode, update, remotes)
  File "C:\Users\Daniel\AppData\Roaming\Python\Python37\site-packages\conans\client\graph\graph_binaries.py", line 176, in _evaluate_node
    self._process_node(node, pref, build_mode, update, remotes)
  File "C:\Users\Daniel\AppData\Roaming\Python\Python37\site-packages\conans\client\graph\graph_binaries.py", line 242, in _process_node
    self._evaluate_cache_pkg(node, package_layout, pref, metadata, remote, remotes, update)
  File "C:\Users\Daniel\AppData\Roaming\Python\Python37\site-packages\conans\client\graph\graph_binaries.py", line 95, in _evaluate_cache_pkg
    assert node.prev, "PREV for %s is None: %s" % (str(pref), metadata.dumps())
AssertionError: PREV for poco/1.9.4:e800054a13ea39bf75eb479d61dc6ce27dd9c321 is None: {"recipe": {"revision": "0", "remote": "conan-center", "properties": {}, "checksums": {"conanmanifest.txt": {"md5": "1c74dc82b5b19f764f9e39ef3530fac9", "sha1": "c0dce2f8e0394be5932455f35d3103877d184bdc"}, "conanfile.py": {"md5": "8315dcc96b131b21d4d64d7f56e73d99", "sha1": "b768bea077de938ea0b95baf916bf6b96101e694"}, "conan_export.tgz": {"md5": "5819fbc0b48456426c49daba6b4d85be", "sha1": "33bca842766631e007b7f5fa21b9770217328b7f"}}}, "packages": {"e800054a13ea39bf75eb479d61dc6ce27dd9c321": {"revision": null, "recipe_revision": null, "remote": null, "properties": {}, "checksums": {}}}}

ERROR: PREV for poco/1.9.4:e800054a13ea39bf75eb479d61dc6ce27dd9c321 is None: {"recipe": {"revision": "0", "remote": "conan-center", "properties": {}, "checksums": {"conanmanifest.txt": {"md5": "1c74dc82b5b19f764f9e39ef3530fac9", "sha1": "c0dce2f8e0394be5932455f35d3103877d184bdc"}, "conanfile.py": {"md5": "8315dcc96b131b21d4d64d7f56e73d99", "sha1": "b768bea077de938ea0b95baf916bf6b96101e694"}, "conan_export.tgz": {"md5": "5819fbc0b48456426c49daba6b4d85be", "sha1": "33bca842766631e007b7f5fa21b9770217328b7f"}}}, "packages": {"e800054a13ea39bf75eb479d61dc6ce27dd9c321": {"revision": null, "recipe_revision": null, "remote": null, "properties": {}, "checksums": {}}}}
```

**Comment by memsharded** [comment]
Hi @sorekz 

I have been able to reproduce, indeed a bug.

It happens to be a complicated issue, an unfortunate alignment of ``short_paths`` + ``conan info --paths`` over packages that doesn't have those paths yet. Doing a ``create`` before the ``info --paths`` works fine.
I am checking some alternatives how to approach this without an extremely ugly hack. In the meantime, I would try to avoid checking those paths in the first place, it always seemed an anti-pattern to me, getting the paths in the cache of packages should be avoided, there are other mechanisms for getting things from the cache (deploy generator, deploy method, imports). Conan might restrict, remove or change this folder access in Conan 2.0

**Comment by Aalmann** [comment]
Hi @memsharded,

do you really mean it is an anti-pattern to get the paths from conan info and one should really use "deploy generator, deploy method, imports", which means one has to copy package contents from A to B?
I completely disagree with that approach due to the fact that it contradicts the base idea of Conan and all other package managers AND it works for all versions prior 1.31. Without using `conan info --paths` there is no official way to get the package paths especially for short paths packages.
How was this handled the versions before? 
And isn't the command line Conan's official API which will stay compatible?  

Best Aalmann

**Comment by memsharded** [comment]
> do you really mean it is an anti-pattern to get the paths from conan info and one should really use "deploy generator, deploy method, imports", which means one has to copy package contents from A to B?

Doing operations on folders directly on the cache might have issues:
- Concurrency. There is very limited support for concurrency on the cache, but there is something, and we might try to improve that in the future.
- There are other mechanisms like the "dirty" flags that are useful to mark incomplete operations while not removing the contents, which is useful for debugging.
- It is easy to break package immutability by accessing the folders directly. Doing such modifications (even involuntary, like creating .pyc or ``__pycache__`` folders as a result of loading files. Conan client explicitly disable that, but it si very possible that any other operation using Python on the cache could leave trailing files in packages that are supposed to be immutable), would be undefined behavior.

No need to worry, it is not going to be removed in Conan 1.X. But it would be good to be aware of these issues and consider other alternatives: what are the operations being done that need raw access to the folders? If it is copying files from there, ``imports`` or ``deploy`` implement exactly that. If it is modifying the files there, that would be breaking packages immutability, not really recommended. If it is running thing from the cache, the ``virtualenv`` generators could help. This is most likely what is going to be recommended in Conan 2.0, and using raw paths from the cache would  be a "last resource" approach.

**Comment by sorekz** [comment]
Hi @memsharded 

We have a generator that collects information from other packages and creates some files based on that.
The packages are not listed in the dependencies of the conanfile itself since they are not real dependencies but calculated by the generator. So we are basically sideloading packages into the cache and copying data from there in case the generator is used.

To achieve this we call `conan info --paths` and use a glob to find the needed files in the export (and sometimes also package) folder.

**Comment by memsharded** [comment]
Fixed in https://github.com/conan-io/conan/pull/8181, will be released in 1.32.1
